src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -565,24 +623,140 @@ private void InterpretDup()
 
         private void InterpretLoadLocal(int index)
         {
+            Debug.Assert(index >= 0);
             _stack.Push(_locals[index]);
         }
 
         private void InterpretStoreLocal(int index)
         {
+            Debug.Assert(index >= 0);
             _locals[index] = PopWithValidation();
         }
 
+        private void InterpretLoadArgument(int index)
+        {
+            Debug.Assert(index >= 0);
+
+            TypeDesc argument = default(TypeDesc);
+
+            if (!_method.Signature.IsStatic)
+            {
+                if (index == 0)
+                    argument = _method.OwningType;
+                else
+                    argument = _method.Signature[index - 1];
+            }
+            else
+            {
+                argument = _method.Signature[index];
+            }
+
+            switch (argument.Category)
+            {
+                case TypeFlags.Boolean:
+                case TypeFlags.Char:
+                case TypeFlags.SByte:
+                case TypeFlags.Byte:
+                case TypeFlags.Int16:
+                case TypeFlags.UInt16:
+                case TypeFlags.Int32:
+                case TypeFlags.UInt32:
+                    _stack.Push(StackItem.FromInt32(GetArgument<int>(index)));
```

Comment:
Is it ok to use `int` to fetch smaller types? I do see anything to ensure that the smaller type will be extended to int properly (ie that the higher bits of `int` are going to have the right values).

Reply:
Updated this to use the actual types instead